pulp_container/tests/functional/api/test_pull_content.py

```diff
@@ -58,126 +42,56 @@ def setUpClass(cls):
-                ContainerContainerDistribution(
-                    **gen_distribution(repository=cls.repo.pulp_href, base_path=repo_dist_path)
-                )
-            )
-            created_resources = monitor_task(distribution_response.task).created_resources
-            distribution = cls.distributions_api.read(created_resources[0])
-            cls.distribution_with_repo = cls.distributions_api.read(distribution.pulp_href)
-            cls.teardown_cleanups.append(
-                (cls.distributions_api.delete, cls.distribution_with_repo.pulp_href)
-            )
-
-            # Step 5.
-            distribution_response = cls.distributions_api.create(
-                ContainerContainerDistribution(
-                    **gen_distribution(
-                        repository_version=cls.repo.latest_version_href,
-                        base_path=repo_ver_dist_path,
-                    )
-                )
-            )
-            created_resources = monitor_task(distribution_response.task).created_resources
-            distribution = cls.distributions_api.read(created_resources[0])
-            cls.distribution_with_repo_version = cls.distributions_api.read(distribution.pulp_href)
-            cls.teardown_cleanups.append(
-                (cls.distributions_api.delete, cls.distribution_with_repo_version.pulp_href)
-            )
-
-            # remove callback if everything goes well
-            stack.pop_all()
-
-    @classmethod
-    def tearDownClass(cls):
-        """Clean class-wide variable."""
-        for cleanup_function, args in reversed(cls.teardown_cleanups):
-            cleanup_function(args)
-
-    def test_api_returns_same_checksum(self):
+        repo = container_repository_factory()
+        remote = container_remote_factory()
+        container_sync(repo, remote)
+        type(self).repo = container_bindings.RepositoriesContainerApi.read(repo.pulp_href)
+        type(self).distribution_with_repo = container_distribution_factory(
+            repository=self.repo.pulp_href
+        )
+        type(self).distribution_with_repo_version = container_distribution_factory(
+            repository_version=self.repo.latest_version_href
+        )
+
+    def test_api_returns_same_checksum(self, container_bindings):
         """Verify that pulp serves image with the same checksum of remote.
 
         1. Call pulp repository API and get the content_summary for repo.
         2. Call dockerhub API and get blobsums for synced image.
         3. Compare the checksums.
         """
         # Get local checksums for content synced from the remote registry
-        checksums = [
-            content["digest"]
-            for content in get_content(self.repo.to_dict())[CONTAINER_CONTENT_NAME]
+        blob_checksums = [
+            c.digest
+            for c in container_bindings.ContentBlobsApi.list(
+                repository_version=self.repo.latest_version_href
+            ).results
```

Comment:
Style nitpick, this would be a bit cleaner if the query was broken out and assigned to a variable.

You don't actually have to fix this. It's merely a suggestion.

Reply:
Also if the primary operation being done here is a lookup, a set comprehension would be better.  But again I don't care that much.